Sources/Data/Microsoft.Psi.Data/Dataset.cs

```diff
@@ -183,6 +252,7 @@ public Session AddSessionFromStore(IStreamReader streamReader, string sessionNam
             var session = new Session(this, sessionName ?? streamReader.Name);
             session.AddStorePartition(streamReader, partitionName);
             this.AddSession(session);
+            this.OnDatasetChanged();

```

Comment:
Since this.AddSession calls OnDatasetChanged, I don't think it's required here. We can do the OnDatasetChanged call at the lowest API levels.